source/whats-new.txt

```diff
@@ -25,18 +25,16 @@ New features of the 4.2 Java driver release include:
 
 - Added Azure and GCP key stores to client-side field level encryption
 - Added Kerberos caching tickets for reuse in multiple authentication requests
-- Added `MongoClients <http://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-sync/com/mongodb/client/MongoClients.html>`_ instances with ``MongoClientSettings`` or ``ConnectionString`` as the configuration
--  The ``explain()`` method can now be used on `find <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-sync/com/mongodb/client/FindIterable.html#explain()>`_ and `aggregate <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-sync/com/mongodb/client/AggregateIterable.html#explain()>`_ commands
-- Added a `JsonObject <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/bson/org/bson/json/JsonObject.html>`_ class to make encoding from and decoding to JSON more efficient by avoiding an intermediate Map representation
-- Added a `BsonRepresentation <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/bson/org/bson/codecs/pojo/annotations/BsonRepresentation.html>`_ annotation that allows ``ObjectId`` BSON values to be represented as a ``String`` in `POJO <https://en.wikipedia.org/wiki/Plain_old_Java_object>`_ classes
-- Added a `Filters.empty() <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Filters.html#empty()>`_ method
+- Added :java-docs:`MongoClients <apidocs/mongodb-driver-sync/com/mongodb/client/MongoClients.html>` instances with ``MongoClientSettings`` or ``ConnectionString`` as the configuration
+-  The ``explain()`` method can now be used on :java-docs:`find <apidocs/mongodb-driver-sync/com/mongodb/client/FindIterable.html#explain()>` and :java-docs:`aggregate <apidocs/mongodb-driver-sync/com/mongodb/client/AggregateIterable.html#explain()>` commands
+- Added a :java-docs:`JsonObject <apidocs/bson/org/bson/json/JsonObject.html>` class to make encoding from and decoding to JSON more efficient by avoiding an intermediate Map representation
+- Added a :java-docs:`BsonRepresentation <apidocs/bson/org/bson/codecs/pojo/annotations/BsonRepresentation.html>` annotation that allows ``ObjectId`` BSON values to be represented as a ``String`` in `POJO <https://en.wikipedia.org/wiki/Plain_old_Java_object>`_ classes
+- Added a :java-docs:`Filters.empty() <apidocs/mongodb-driver-core/com/mongodb/client/model/Filters.html#empty()>` method
```

Comment:
Issue: as these links correspond to specific versions of the driver. I believe they should be hardcoded as before.